lib/minify.js

```diff
@@ -291,11 +288,21 @@ async function minify(files, options, _fs_module) {
         toplevel.print(stream);
         result.code = stream.get();
         if (options.sourceMap) {
-            if(options.sourceMap.asObject) {
-                result.map = options.format.source_map.get().toJSON();
-            } else {
-                result.map = options.format.source_map.toString();
-            }
+            Object.defineProperty(result, "map", {
+                configurable: true,
+                enumerable: true,
+                get() {
+                    const map = options.format.source_map.getEncoded();
+                    return (result.map = options.sourceMap.asObject ? map : JSON.stringify(map));
+                },
+                set(value) {
+                    Object.defineProperty(result, "map", {
+                        value,
+                        writable: true,
+                    });
+                }
+            });
+            result.decodedMap = options.format.source_map.getDecoded();
```

Comment:
```suggestion
            result.decoded_map = options.format.source_map.getDecoded();
```

The public API uses underscores throughout, we shouldn't break it here. I know `sourceMap` is camelCase but that's something that should be changed in the future. (or at least accept sourceMap and source_map).